### ISSUE TITLE
Tween created in _ready does not work if used later

### ISSUE THREAD

**Issue by lyuma** [godotengine/godot#52147]
### Godot version

4.0.dev 6e87d62

### System information

Windows 10, NVidia, Vulkan

### Issue description

I have an autoload manager which creates a Tween and will use it to fade the screen in and out.

This code worked in Godot 3, but after making a best effort attempt to port the Tween code to the new API in Godot 4, it seems to give confusing errors about Tweeners and fail unless I recreate the object every time. I have also triple-checked and Autoloaded class are definitely in the scene tree. It shows the same error if I use with get_tree() or without.

Here is an excerpt of the output if I attempt to run the attached project. The "is_valid" and "fade_manager" prints are mine. The others come from the C++ Tween class itself.
```
Created a new Tween: is_valid=False is_running=True
ERROR: Tween started, but has no Tweeners.
   at: (scene\animation\tween.cpp:221)
Tween: is_valid=True is_running=True
ERROR: Tween was created outside the scene tree, can't use Tweeners.
   at: (scene\animation\tween.cpp:56)
ERROR: fade_manager.gd: Failed to create a tween!
   at: VariantUtilityFunctions::push_error (core\variant\variant_utility.cpp:607)
ERROR: Tween invalid, can't play.
   at: (scene\animation\tween.cpp:120)
```

A random guess is maybe tweens are garbage collected after some delay, but I can't find any mention of this in the updated docs in doc/classes/Tween.xml. It's possible I misread it.

The thing is if I start the fade immediately inside of _ready, it actually seems to work if I do it there, so there's something about timing.

### Steps to reproduce

1. Play the attached project.
2. Wait a second for the simulated 1 second loading time is finished.
3. It should fade to black, but prints an error and hits fallback code to instantly set the screen to black.
4. Edit fade_manager.gd and uncomment the call to "_recreate_tween" on line 29
5. Try again. The Tween will now work.

### Minimal reproduction project

[TweenAutoload.zip](https://github.com/godotengine/godot/files/7063754/TweenAutoload.zip)

**Comment by KoBeWi** [comment]
The stop bug you mentioned (it was actually a bug) was fixed in #57353
The `is_valid()` bug was fixed in #51237

I still don't know why would you want to re-use Tweens, but it's now fully possible. You just need to stop them right after creating and right after finishing.

**Comment by razcore-rad** [comment]
> The stop bug you mentioned (it was actually a bug) was fixed in #57353 The `is_valid()` bug was fixed in #51237
> 
> I still don't know why would you want to re-use Tweens, but it's now fully possible. You just need to stop them right after creating and right after finishing.

Consider this:

1.
```gdscript
var tween: Tween = null

func jump_to(target : Vector2):
	if tween != null and tween.is_valid():
		return

	tween = create_tween()
	# do tween stuff
```

vs.

2.
```gdscript
@onready var tween: Tween = create_tween()

func jump_to(target : Vector2):
	tween.stop()
	# do tween stuff
```

I'm settling with:

3.
```gdscript
@onready var tween: Tween = create_tween()

func _ready() -> void:
	# have to stop it manually after creation otherwise Godot complains
	tween.kill()

func jump_to(target : Vector2):
	tween.kill()
	tween = create_tween()
	# do tween stuff
```

Even though I think 2. would be the developer experience.
